jwt.js

```diff
@@ -202,12 +206,28 @@ function fastifyJwt (fastify, options, next) {
 
   next()
 
+  function getVerifier (options, globalOptions) {
+    const useGlobalOptions = globalOptions ?? options === verifierConfig.options
+    // Use global verifier if using global options with static key
+    if (useGlobalOptions && hasStaticPublicKey) return verifier
+    // Only cache verifier when using default options (except for key)
+    if (useGlobalOptions && options.key && typeof options.key === 'string') {
+      let verifier = validatorCache.get(options.key)
+      if (!verifier) {
+        verifier = createVerifier(options)
+        validatorCache.set(options.key, verifier)
+      }
+      return verifier
+    }
+    return createVerifier(options)
+  }
+
   function decode (token, options) {
     assert(token, 'missing token')
 
     let selectedDecoder = decoder
 
-    if (options && typeof options !== 'function') {
+    if (options && options !== decodeOptions && typeof options !== 'function') {
```

Comment:
a new decoder was also being created on every request (see line 478), this ensures we will use the global decoder if the default/global settings are used.